### DISCUSSION TITLE
debug `type` not recognized

### DISCUSSION THREAD

**Discussion by subratkatiyar** [eclipse-theia/theia#11789] [Q&A]
Hi,
I was experimenting with custom debug adapters for Theia. I have an extension which provides a custom debug type. I get an error message that the debug type is not recognized. Interesting part being, debugging works completely fine with this configuration.

![image](https://user-images.githubusercontent.com/43543908/196859324-0f1ca07e-0fa4-49a0-81c6-0ea725407509.png)

Has anyone faced similar issue? Is this a bug or am I doing something wrong.

Backend.
![image](https://user-images.githubusercontent.com/43543908/196860119-40ec4fa8-fa46-415a-9a7a-53137862b4f0.png)

**Comment by vince-fugnitto** [comment]
@DesmondWasHere thank you for the discussion, I haven't seen the issue myself, would you be able to share your custom debug adapter extension?

The message comes from the following (`type` being a mandatory field):

https://github.com/eclipse-theia/theia/blob/7e753f459e46a6c25695c9f80647d028c9d44af6/packages/debug/src/node/vscode/vscode-debug-adapter-contribution.ts#L147-L155

You could also debug in this area (using `Launch Browser Backend & Frontend` or `Launch Electron Backend & Frontend` debug configurations) to see what might be happening.

**Reply by subratkatiyar** [reply]
I will try to debug this issue, in my free time. I have added the steps to reproduce, if you want to try it out.

**Comment by subratkatiyar** [comment]
@vince-fugnitto 
Steps to reproduce:
1. Use the **VS Code yo code generator** to generate a simple hello-world extension.
`npm install -g yo generator-code`
`yo code`
Configuration.
![image](https://user-images.githubusercontent.com/43543908/197132807-d179bb25-3277-45e0-9368-26fc16cb29f6.png)

2.  In package.json contribute a debugger.
![image](https://user-images.githubusercontent.com/43543908/197133151-e8a3c776-8b25-4a79-a5ff-72dc8e7c1b86.png)

3. Package the extension(.vsix) and install in Theia. [Edit the readme file (a pre-requisite for packaging vscode-extensions)]
`npm install -g vsce`
`vsce package`
The following warnings can be ignored. Simply say **Yes** and continue.
![image](https://user-images.githubusercontent.com/43543908/197135089-c09dd1a8-37bb-4a2b-bb46-8ff2e7ee1dd7.png)


You can also use this pre-packaged extension to test.
[vscode-type-check-0.0.1.zip](https://github.com/eclipse-theia/theia/files/9836465/vscode-type-check-0.0.1.zip)
There is a hello world command, which can be used to test if the extension was loaded properly.
![image](https://user-images.githubusercontent.com/43543908/197134279-988ee143-23e6-4eba-8a66-c127e7dceebf.png)
